src/components/AddNewTask.tsx

```diff
@@ -0,0 +1,66 @@
+import { useState } from "react";
+import { ExpandLess, ExpandMore } from "@mui/icons-material";
+import { useTasks } from "../context/TaskContext";
+import { useForm } from "react-hook-form";
+
+export const AddNewTask = () => {
+  const [isCollapsed, setIsCollapsed] = useState(false);
+  const toggleCollapse = () => setIsCollapsed(!isCollapsed);
+  const { addTask } = useTasks();
+
+  const { register, handleSubmit, formState: { errors }, reset } = useForm({
+    defaultValues: {
+      taskName: '',
+      taskDescription: ''
+    }
+  });
+
+  const onSubmit = (data: { taskName: string; taskDescription: string }) => {
+    addTask({ id: crypto.randomUUID(), name: data.taskName, description: data.taskDescription, status: 'todo', createdAt: new Date() });
+    reset();
+  };
+
+  return (
+    <div className={`border-2 border-primary bg-secondary rounded-lg p-4 flex flex-col gap-4`}>
+      <div className="flex justify-between">
+        <p className="text-xl font-semibold">New task</p>
+        {isCollapsed ? (
+            <ExpandMore onClick={toggleCollapse}  className="cursor-pointer"  />
```

Comment:
you could shorten this code by directly passing setIsCollapsed method since that's the only purpose for toggleCollapse method